File: apps/api/src/common/geo/macedonia-bounds.ts
/**
 * Shared North Macedonia bounding box, used everywhere the server must decide whether
 * a coordinate is "inside the country" (home-location onboarding, report submit,
 * event check-in, and home-location onboarding).
 *
 * This is intentionally a coarse rectangle (matches the mobile onboarding map and the
 * report geofence) - it is a server-side guard, not a precise polygon. Keep this the
 * single source of truth so the client and server never drift.
 */
export const MACEDONIA_BOUNDS = {
  minLat: 40.8,
  maxLat: 42.4,
  minLng: 20.4,
  maxLng: 23.1,
} as const;

/** Returns true when the WGS84 coordinate falls inside the North Macedonia bounding box. */
export function isWithinMacedonia(latitude: number, longitude: number): boolean {
  return (
    Number.isFinite(latitude) &&
    Number.isFinite(longitude) &&
    latitude >= MACEDONIA_BOUNDS.minLat &&
    latitude <= MACEDONIA_BOUNDS.maxLat &&
    longitude >= MACEDONIA_BOUNDS.minLng &&
    longitude <= MACEDONIA_BOUNDS.maxLng
  );
}

File: apps/mobile/packages/chisto_infrastructure/lib/core/location/macedonia_bounds.dart
/// Single source of truth for Macedonia bounds (axis-aligned box).
/// Matches server [MACEDONIA_BOUNDS] in apps/api.
class MacedoniaBounds {
  static const double minLat = 40.8;
  static const double maxLat = 42.4;
  static const double minLng = 20.4;
  static const double maxLng = 23.1;

  static bool contains(double latitude, double longitude) {
    return latitude.isFinite &&
        longitude.isFinite &&
        latitude >= minLat &&
        latitude <= maxLat &&
        longitude >= minLng &&
        longitude <= maxLng;
  }
}

bool isWithinMacedonia(double latitude, double longitude) =>
    MacedoniaBounds.contains(latitude, longitude);
